src/parquet/column/page.h

```diff
@@ -95,6 +95,21 @@ class DataPage : public Page {
   std::string min_;
 };
 
+class CompressedDataPage : public DataPage {
+ public:
+  CompressedDataPage(const std::shared_ptr<Buffer>& buffer, int32_t num_values,
```

Comment:
I was a bit confused that we do not have to have the compression codec here. For the moment it is ok, as it will not be used. Once we use the CompressedDataPage for reading, it might be worth it (but it seems that there is not that much freedom in having different compression codec inside a single column chunk.)